src/sentry/static/sentry/app/components/events/interfaces/frameRegisters.jsx

```diff
@@ -0,0 +1,139 @@
+import {Flex, Box} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import Tooltip from 'app/components/tooltip';
+import {t} from 'app/locale';
+import {defined, objectToArray} from 'app/utils';
+
+const REGISTER_VIEWS = [t('Hexadecimal'), t('Numeric')];
+
+export class RegisterValue extends React.Component {
+  static propTypes = {
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      view: 0,
+    };
+  }
+
+  toggleView = () => {
+    this.setState({
+      view: (this.state.view + 1) % REGISTER_VIEWS.length,
+    });
+  };
+
+  formatValue = value => {
+    try {
+      let parsed = typeof value === 'string' ? parseInt(value, 16) : value;
+      if (isNaN(parsed)) return value;
+
+      switch (this.state.view) {
+        case 1:
+          return String(parsed);
+        case 0:
+        default:
+          return '0x' + ('0000000000000000' + parsed.toString(16)).substr(-16);
+      }
+    } catch (e) {
+      return value;
+    }
+  };
+
+  render() {
+    return (
+      <InlinePre>
+        <FixedWidth>{this.formatValue(this.props.value)}</FixedWidth>
+        <Tooltip title={REGISTER_VIEWS[this.state.view]} onClick={this.toggleView}>
+          <Toggle className="icon-filter" />
+        </Tooltip>
+      </InlinePre>
+    );
+  }
+}
+
+class FrameRegisters extends React.Component {
+  static propTypes = {
+    data: PropTypes.object.isRequired,
+  };
+
+  // make sure that clicking on the registers does not actually do
+  // anything on the containing element.
+  preventToggling = evt => {
+    evt.stopPropagation();
+  };
+
+  render() {
+    let registers = objectToArray(this.props.data).filter(register =>
+      defined(register[1])
+    );
+
+    return (
+      <div className="registers">
```

Comment:
Since we're using `emotion` everywhere else, can we use it here too? I believe the CSS below should work.

I've been tossing around this pattern for styled components:

```javascript
const FrameRegisters = styled(
  class FrameRegistersComponent extends React.Component {
    static propTypes = {
      title: PropTypes.node,
    };

    render() {
      let {className, children} = this.props;
      return (
        <div className={className}>
          {children}
        </div>
      );
    }
  }
)`
   border-top: 1px solid @trim;
   padding-top: 10px;

   .box-clippable:first-child & {
      border-top: none;
      padding-top: 0;
   }
`;
```